featurebyte/service/feature_store_warehouse.py

```diff
@@ -304,16 +407,31 @@ async def get_table_details(
         -------
         TableDetails
         """
-
-        db_session = await self.session_manager_service.get_feature_store_session(
-            feature_store=feature_store
+        if refresh:
+            table_details_cache[feature_store.id].clear()
+        credentials = await self.session_manager_service.credential_service.get_credentials(
+            user_id=self.session_manager_service.user.id, feature_store=feature_store
         )
-        try:
-            return await db_session.get_table_details(
-                database_name=database_name, schema_name=schema_name, table_name=table_name
+        key = hashkey(
+            credentials.id if credentials else None,
+            database_name,
+            schema_name,
+            table_name,
+        )
+        result = table_details_cache[feature_store.id].get(key)
+        if result is None:
```

Comment:
nit: The logic for checking cache existence and read/update seems repetitive across different methods. Alternatively we could have a method decorator that accepts a cache object and handles the caching logic. This keeps the method bodies focused on their core functionality without being aware of caching details.